packages/jest-config/src/normalize.js

```diff
@@ -371,6 +376,11 @@ function normalize(config: InitialConfig, argv: Object = {}) {
     }
   }
 
+  if (config.testRegex && (!config.testGlob || config.testGlob.length === 0)) {
+    // Prevent default glob conflicting with any explicit `testRegex` value
+    newConfig.testGlob = [];
+  }
```

Comment:
Wait, I don't really understand this. Why are you setting testGlob to an empty array if testGlob is empty?